packages/react-ai/src/components/AIConversation/views/Controls/FormControl.tsx

```diff
@@ -165,8 +172,23 @@ export const FormControl: FormControl = () => {
   const ref = React.useRef<HTMLFormElement | null>(null);
   const controls = React.useContext(ControlsContext);
   const [composing, setComposing] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
 
   const submitMessage = async () => {
+    const hasInput =
+      !!input?.text || (input?.files?.length && input?.files?.length > 0);
+    // Prevent double submission and empty submission
+    if (isSubmitting || !hasInput) {
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    // Clear the attachment errors when submitting
+    // because the errors are not actually preventing the submission
+    // but rather notifying the user that certain files were not attached and why they weren't
+    setError?.(undefined);
```

Comment:
Should this happen last since it's a cleanup behavior? Might make sense to move the call to `ref.current?.reset()` to the bottom of the function body as well